torch/_higher_order_ops/scan.py

```diff
@@ -39,10 +40,7 @@ def wrap_combine_fn_flat(
     carry = pytree.tree_unflatten(args[:num_init_leaves], spec_init)
     xs = pytree.tree_unflatten(args[num_init_leaves:], spec_xs)
     carry, combined = combine_fn(carry, xs)
-    carry_flat = pytree.tree_leaves(carry)
-    combined_flat = pytree.tree_leaves(combined)
-    assert num_init_leaves == len(carry_flat)
-    return [*carry_flat, *combined_flat]
+    return [carry, combined]
```

Comment:
it might not always be semantic preserving if users actually return a tuple, can just `return combine_fn(carry, xs)`.

Reply:
Done